components/gitpod-cli/cmd/ports/ports-list.go

```diff
@@ -0,0 +1,75 @@
+// Copyright (c) 2022 Gitpod GmbH. All rights reserved.
+// Licensed under the GNU Affero General Public License (AGPL).
+// See License-AGPL.txt in the project root for license information.
+
+package ports
+
+import (
+	"context"
+	"fmt"
+	"os"
+	"time"
+
+	"github.com/gitpod-io/gitpod/gitpod-cli/pkg/supervisor"
+	"github.com/gitpod-io/gitpod/supervisor/api"
+	"github.com/spf13/cobra"
+
+	"github.com/olekukonko/tablewriter"
+)
+
+func ListPortsCmd(cmd *cobra.Command, args []string) {
+	ctx, cancel := context.WithTimeout(context.Background(), 5*time.Second)
+	defer cancel()
+
+	conn := supervisor.Dial()
+	client := api.NewStatusServiceClient(conn)
+
+	ports := supervisor.GetPortsList(ctx, client)
+
+	if len(ports) == 0 {
+		fmt.Println("No ports detected")
+		return
+	}
+
+	table := tablewriter.NewWriter(os.Stdout)
+	table.SetHeader([]string{"Port", "Status", "URL", "Name & Description"})
```

Comment:
It's unfortunate that the URL makes the table hard to read in terminal split mode.

<img width="1392" alt="Screenshot 2022-05-31 at 17 06 29" src="https://user-images.githubusercontent.com/2318450/171292195-960bc1c1-b217-42e3-a7df-2a418e1366ea.png">

At the same time, the URL is handy. Also, the description might be long so I cannot think of a better approach.

Reply:
Hopefully, it will be shorter when under the `gitpod.io` domain.

For future PRs, by adding arguments to this command, we can filter the columns to show and display them in a different format (listing one entry below another, for example) :)